src/v/debug_bundle/debug_bundle_service.cc

```diff
@@ -469,36 +593,160 @@ service::build_rpk_arguments(job_id_t job_id, debug_bundle_parameters params) {
     return rv;
 }
 
+ss::future<> service::cleanup_previous_run() const {
+    if (_rpk_process == nullptr) {
+        co_return;
+    }
+
+    auto& debug_bundle_file = _rpk_process->output_file_path().native();
+    if (co_await ss::file_exists(debug_bundle_file)) {
+        vlog(
+          lg.debug,
+          "Cleaning up previous debug bundle run {}",
+          debug_bundle_file);
+        co_await ss::remove_file(debug_bundle_file);
+    }
+
+    co_await remove_kvstore_entry();
+}
+
+ss::future<> service::remove_kvstore_entry() const {
+    co_await _kvstore->remove(
+      storage::kvstore::key_space::debug_bundle,
+      bytes::from_string(debug_bundle_metadata_key));
+}
+
+ss::future<> service::set_metadata(job_id_t job_id) {
+    auto& debug_bundle_file = _rpk_process->output_file_path();
+    auto run_successful = [this]() {
+        auto wait_status = _rpk_process->get_wait_result();
+        auto* exited = std::get_if<ss::experimental::process::wait_exited>(
+          &wait_status);
+        return exited != nullptr && exited->exit_code == 0;
+    }();
+
+    bytes sha256_checksum{};
+    if (run_successful) {
+        if (!co_await ss::file_exists(debug_bundle_file.native())) {
+            vlog(
+              lg.warn,
+              "Debug bundle file {} does not exist post successful run, cannot "
+              "set "
+              "metadata",
+              debug_bundle_file);
+            co_return;
+        }
+        sha256_checksum = co_await calculate_sha256_sum(
+          debug_bundle_file.native());
+    }
+
+    metadata md(
+      _rpk_process->created_time(),
+      _rpk_process->cout().copy(),
+      _rpk_process->cerr().copy(),
+      job_id,
+      debug_bundle_file,
+      std::move(sha256_checksum),
+      _rpk_process->get_wait_result());
+
+    iobuf buf;
+    serde::write(buf, std::move(md));
```

Comment:
kvstore isn't really designed to handle large data. so storing stdout/stderr is a bit concerning here.

Reply:
how big is too big? I want to say this is a few hundred bytes in the common case, but I'm not too sure.

Reply:
what if go crashed and produced a large backtrace? or whatever there is a bug in the collector and it dumped a bunch of stuff out.

>how big is too big?

yeh good question--it is stored in ram for the lifetime of the key, and there is no option not to store it in ram.

Reply:
could store it on disk as a file with the same key

Reply:
So I was planning on running `rpk debug bundle` with the `--verbose` flag which would probably increase the amount of data in stdout/stderr, which I guess could produce a bit of information.

Why keep around stdout/stderr?

In case a previous run of `rpk debug bundle` failed and the broker reset.  Maybe we'd like to know why it failed.  Or on success, if there was anything interesting in the verbose output that may help with debugging a problem.

Why kvstore?

Originally I was going to output a json metadata file with all this information in it, but as reminded about the existence of the kvstore, so switched to doing that.

So to Oren's question, how big is too big?

Reply:
> So to Oren's question, how big is too big?

we don't have any formal guidelines, but the intention of the kvstore in its current state is to store small metadata.

seems you could keep the metadata in the kvstore, and the stdout/stderr on disk with the same key name?

Reply:
Happy to move it out of the kvstore and onto disk.  If doing so I wonder if it should be JSON or it's fine to remain in a serde format.

Reply:
> seems you could keep the metadata in the kvstore, and the stdout/stderr on disk with the same key name?

Yeah that's probably fine something like `<uuid>.stdout` and `<uuid>.stderr`.

Reply:
>If doing so I wonder if it should be JSON or it's fine to remain in a serde format.

binary is probably fine. its presumably ascii or utf8?